src/lib/builtins/string.rs

```diff
@@ -728,6 +729,60 @@ pub fn match_all(this: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultV
     regexp_match_all(&re, ctx.value_to_rust_string(this))
 }
 
+/// Return a String where the matches of a pattern in the original string is replaced with a given
+/// replacement.
+/// <https://tc39.es/ecma262/#sec-string.prototype.replace>
+pub fn replace(this: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultValue {
+    // First we get it the actual string a private field stored on the object only the engine has access to.
+    // Then we convert it into a Rust String
+    let this_str: &str = &ctx.value_to_rust_string(this);
+
+    // Make a Rust Regex from the first argument
+    let pattern = get_argument(args, 0);
+    if pattern.is_undefined() {
+        return Ok(to_value(this_str));
+    }
+
+    let pattern_re = if let Some(re_internal_state) = pattern.get_internal_state() {
+        if let Some(pattern_jsre) = re_internal_state.downcast_ref::<RegExp>() {
+            pattern_jsre.get_matcher().clone()
+        } else {
+            let pattern_str = &ctx.value_to_rust_string(&pattern);
+            regex::Regex::new(&regex::escape(pattern_str)).unwrap()
+        }
+    } else {
+        let pattern_str = &ctx.value_to_rust_string(&pattern);
+        regex::Regex::new(&regex::escape(pattern_str)).unwrap()
+    };
+
+    // Make a replacement closure from the second argument
+    let replacement = get_argument(args, 1);
+
+    let replacement_cls = if replacement.is_function() {
+        Box::new(|caps: &Captures| {
+            let m = caps.get(0).unwrap().as_str();
+            let args = vec![to_value(m)];
+            let rs_value = ctx
+                .call(&replacement, &undefined(), args)
+                .unwrap_or_else(|_| undefined());
+            ctx.value_to_rust_string(&rs_value)
+        }) as Box<dyn FnMut((&Captures)) -> (String)>
```

Comment:
Remove the parentheses, so it becomes

```suggestion
        }) as Box<dyn FnMut(&Captures) -> String>
```